### ISSUE TITLE
ms-vsliveshare-vsliveshare fails to build due to dotnet globalization error

### ISSUE THREAD

**Issue by ppenguin** [NixOS/nixpkgs#173145]
### Steps To Reproduce
1. pull in `unstable.vscode-extensions.ms-vsliveshare.vsliveshare` from `home-manager` `programs.vscode.extensions`
2. observe that the build fails (this appears to be a regression since approximately beginning of May?), see build log below.

### Build log
```
@nix { "action": "setPhase", "phase": "unpackPhase" }
unpacking sources
unpacking source archive /nix/store/v341z9d3h8f9kxx1k84r5wh60akz3ysw-dotnet-sdk-3.1.415-linux-x64.tar.gz
source root is .
setting SOURCE_DATE_EPOCH to timestamp 1635274917 of file ./sdk/3.1.415/Sdks/NuGet.Build.Tasks.Pack/CoreCLR/NuGet.Build.Tasks.Pack.dll
@nix { "action": "setPhase", "phase": "patchPhase" }
patching sources
@nix { "action": "setPhase", "phase": "configurePhase" }
configuring
no configure script, doing nothing
@nix { "action": "setPhase", "phase": "buildPhase" }
building
no Makefile, doing nothing
@nix { "action": "setPhase", "phase": "installPhase" }
installing
@nix { "action": "setPhase", "phase": "fixupPhase" }
post-installation fixup
rewriting symlink /nix/store/zaak5j4w1hm1cdl4iijks8afgcb51gz0-dotnet-sdk-3.1.415/bin/dotnet to be relative to /nix/store/zaak5j4w1hm1cdl4iijks8afgcb51gz0-dotnet-sdk-3.1.415
strip is /nix/store/bg35nfwn6zd616facdywiysgpprfvsji-gcc-wrapper-11.3.0/bin/strip
stripping (with command strip and flags -S) in /nix/store/zaak5j4w1hm1cdl4iijks8afgcb51gz0-dotnet-sdk-3.1.415/bin 
patching script interpreter paths in /nix/store/zaak5j4w1hm1cdl4iijks8afgcb51gz0-dotnet-sdk-3.1.415
checking for references to /build/ in /nix/store/zaak5j4w1hm1cdl4iijks8afgcb51gz0-dotnet-sdk-3.1.415...
@nix { "action": "setPhase", "phase": "installCheckPhase" }
running install tests
Process terminated. Couldn't find a valid ICU package installed on the system. Set the configuration flag System.Globalization.Invariant to true if you want to run with no globalization support.
   at System.Environment.FailFast(System.String)
   at System.Globalization.GlobalizationMode.GetGlobalizationInvariantMode()
   at System.Globalization.GlobalizationMode..cctor()
   at System.Globalization.CultureData.CreateCultureWithInvariantData()
   at System.Globalization.CultureData.get_Invariant()
   at System.Globalization.CultureInfo..cctor()
   at System.String.ToLowerInvariant()
   at Microsoft.DotNet.PlatformAbstractions.RuntimeEnvironment.GetArch()
   at Microsoft.DotNet.PlatformAbstractions.RuntimeEnvironment..cctor()
   at Microsoft.DotNet.PlatformAbstractions.RuntimeEnvironment.GetRuntimeIdentifier()
   at Microsoft.DotNet.Cli.MulticoreJitProfilePathCalculator.CalculateProfileRootPath()
   at Microsoft.DotNet.Cli.MulticoreJitActivator.StartCliProfileOptimization()
   at Microsoft.DotNet.Cli.MulticoreJitActivator.TryActivateMulticoreJit()
   at Microsoft.DotNet.Cli.Program.Main(System.String[])
/nix/store/crpnj8ssz0va2q0p5ibv9i6k6n52gcya-stdenv-linux/setup: line 1373:   125 Aborted                 (core dumped) $out/bin/dotnet --info
```

### Metadata
Please run `nix-shell -p nix-info --run "nix-info -m"` and paste the result.

```console
> nix-shell -p nix-info --run "nix-info -m"
 - system: `"x86_64-linux"`
 - host os: `Linux 5.17.5, NixOS, 21.11 (Porcupine)`
 - multi-user?: `yes`
 - sandbox: `yes`
 - version: `nix-env (Nix) 2.8.0`
 - channels(jeroen): `"home-manager"`
 - channels(root): `"nixos-21.11"`
 - nixpkgs: `/nix/var/nix/profiles/per-user/root/channels/nixos`
```

**Comment by GGG-KILLER** [comment]
Seems to be caused by #173188 which was fixed by #172335

**Comment by ppenguin** [comment]
Working again (see above).
/close
